internal/build/lifecycle_executor.go

```diff
@@ -102,6 +103,8 @@ type LifecycleOptions struct {
 	SBOMDestinationDir              string
 	CreationTime                    *time.Time
 	Keychain                        authn.Keychain
+	DetectOnly                      bool
+	FetchOptions                    image.FetchOptions
```

Comment:
Is this used? And if so, what is it for?

Reply:
With the above change, these wont be needed anymore. Removed